### ISSUE TITLE
[Disscussion] Keep langchiango  lightweight enouph for everyone

### ISSUE THREAD

**Issue by edocevol** [tmc/langchaingo#259]
Honestly speaking, I think we should ensure that this package is lightweight enough for everyone to use in production environments. We can implement different implementations based on the interfaces defined in this package, and these implementations should be as independent as possible from the current package. All langchain components should also follow this principle.

**Comment by tmc** [comment]
I completely agree. We've allowed a few too many third party dependencies to creep in -- can folks please file issues or leave comments here on places where we should be programming to interfaces instead so that we can drop third party dependencies from the main tree? (happy to put those impl-specific implementation modules into the tree with separate go.mod files).
